preprocess.py

```diff
@@ -98,6 +101,18 @@ def cut_to_same_size(data_1, data_2):
         The two original arrays, the longer one cut at the end to the length of
         the shorter one so that both arrays have the same length.
     """
+    length = 0
+    if( len(data_1) > len(data_2)):
```

Comment:
another solution would be finding the minimum of lengths of both strings (using `min` function)